js/Heap.js

```diff
@@ -22,14 +20,25 @@ import { checkDisplayNamePlugin } from './util/checkDisplayNames';
 import { withReactNavigationAutotrack } from './autotrack/reactNavigation';
 import { bailOnError } from './util/bailer';
 import { getContextualProps } from './util/contextualProps';
+import Config from 'react-native-config'
+import { createClient } from '@segment/analytics-react-native'
 
 const flatten = require('flat');
-const RNAnalytics = NativeModules.RNAnalytics;
+
+const client = createClient({
+  writeKey: Config.SEGMENT_API_KEY,
+  debug: true,
+  trackAppLifecycleEvents: true,
+  flushInterval: 60000, // 60 seconds
+})
+
+// track only for production build as we can change environments in other builds
+const shouldTrack = Config.ENV === 'production'
```

Comment:
💬 Hmmm. I thought it is still useful to track events from admins who are viewing member data. Do you remember the original reason we stopped tracking in non-prod environments?

Reply:
@Wittiest 

In our app we have this condition:

```
  shouldEnableSegment: () => {
    // only beta and production builds use production segment API key
    if (App.build.beta || App.build.production) {
      return Apollo.endpointEnv === 'production'
    }

    // we don't need to enable it for development endpoint as we have different data locally
    return Apollo.endpointEnv === 'staging'
  },
```

The issue was that we've used staging endpoint with Beta app, so user's data has been updated with data from staging

Reply:
@Wittiest 
I can add saving  `Apollo.endpointEnv` to AsyncStorage and getting it here, so we have the same logic if it's necessary, wdyt?

Reply:
@storm2513 Is there any way to pass a config from the mobile repo into heap?

Reply:
@Wittiest updated